src/item_location.cpp

```diff
@@ -359,6 +368,12 @@ class item_location::impl::item_on_person : public item_location::impl
         void remove_item() override {
             who.remove_item( *what );
         }
+
+        void remove_item_keep_contents() override {
+            for( auto &content : who.remove_item( *what ).contents ) {
+                who.i_add_or_drop( content );
```

Comment:
Might be a little cleaner to delegate to `character::i_rem_keep_contents()` here.

Reply:
The reason I didn't do this is because `i_rem_keep_contents` takes an `int pos` rather than an item reference, which I'm not sure how to obtain. Perhaps I could refactor `i_rem_keep_contents` to take an item reference?

Reply:
what you were looking for is `character::get_item_position( const item * )`